plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/WatchManager.java

```diff
@@ -61,22 +62,26 @@ public static WatchManager getInstance() {
 		return Holder.instance;
 	}
 	
-	public 	void stopWatch(IProject project) {
-		if(watches.containsKey(project)){
-			watches.remove(project).stop();
+	public void stopWatch(IProject project) {
+		Map<String, IWatcher>  watchList = watches.remove(project);
+		if(watchList != null) {
+			watchList.values().forEach(w->w.stop());
 		}
 	}
 
 	public void startWatch(final IProject project) {
-		if(!watches.containsKey(project)){
-			startWatch(project, 0, 0);
+		ConcurrentMap<String, IWatcher> kindMap = new ConcurrentHashMap<>();
+		if(watches.putIfAbsent(project, kindMap) == null){
+			final Connection conn = ConnectionsRegistryUtil.getConnectionFor(project);
```

Comment:
I will add a check here but if it's null then we have a problem.  It would mean that someone created a project without using our factory